azure-pipelines.yml

```diff
@@ -102,8 +102,47 @@ jobs:
                $(Slack.URL)
         condition: and(failed(), eq(variables['Build.SourceBranchName'], 'master'))
 
+  - job: Windows_signing
+    # Signing is a separate job so that we can make sure that we only sign on releases.
+    # Since the release check is run on Linux, we do not have access to that information
+    # in the regular Windows step.
+    dependsOn: [ "Windows", "Linux" ]
+    pool:
+      name: 'windows-pool'
+    condition: and(succeeded(), eq(dependencies.Linux.outputs['release.has_released'], 'true'))
+    variables:
+      unsigned-installer: $[ dependencies.Windows.outputs['publish.artifact-unsigned-windows-installer'] ]
+    steps:
+      - checkout: self
+        persistCredentials: true
+      - bash: |
+          set -euo pipefail
+          echo "##vso[task.setvariable variable=release_tag]$(cat VERSION)"
+        name: release
+      - task: DownloadPipelineArtifact@0
+        inputs:
+          artifactName: $(unsigned-installer)
+          targetPath: $(Build.StagingDirectory)/
+      - bash: |
+          set -euo pipefail
+          INSTALLER=daml-sdk-$(release_tag)-windows.exe
+          mv "$(Build.StagingDirectory)/$(unsigned-installer)" "$(Build.StagingDirectory)/$INSTALLER"
+          chmod +x "$(Build.StagingDirectory)/$INSTALLER"
+          echo "$SIGNING_KEY" | base64 -d > signing_key.pfx
+          MSYS_NO_PATHCONV=1 signtool.exe sign '/f' signing_key.pfx '/fd' sha256 '/tr' "http://timestamp.digicert.com" '/v' "$(Build.StagingDirectory)/$INSTALLER"
```

Comment:
Feels a bit weird to be using an `http` call in a signing phase. Any chance you could turn this into an `https` call?

Reply:
`curl https://timestamp.digicert.com` doesn't seem to work, so I suppose that's off the table. Still feels weird though.

Reply:
Yeah I was rather surprised by this but I copied it from the official [Digicert documentation](https://www.digicert.com/code-signing/signcode-signtool-command-line.htm). I took a short look at [RFC 3161](https://www.ietf.org/rfc/rfc3161.txt) which defines the timestamping protocol and it looks like that has a custom signing mechanism so presumably that’s why using HTTP is reasonable here.